Update azure functions docs (#8176)
apollographql/apollo-server: docs/source/deployment/azure-functions.mdx

import { startServerAndCreateHandler } from '@as-integrations/azure-functions';

// Define the context type
type MyContext = {
  user: string | null;
  isAuthenticated: boolean;
};

// Context creation function
async function createContext(
  req: HttpRequest,
  _context: InvocationContext,
  _body: any,
): Promise<MyContext> {
  const authHeader = req.headers.get('authorization');

  return {
    user: authHeader ? authHeader : null,
    isAuthenticated: authHeader !== null,
  };
}

const typeDefs = `#graphql
  type Query {
    hello: String
    me: User
  }

  type User {
    id: ID!
    name: String!
  }
`;

const resolvers = {
  Query: {
    hello: () => 'world',
    me: (_: any, _args: any, ctx: MyContext) => {
      if (!ctx.isAuthenticated) {
        throw new Error('Not authenticated');
      }
      return { id: '1', name: ctx.user };
    },
  },
};

const server = new ApolloServer<MyContext>({
  typeDefs,
  resolvers,
});

app.http('graphql', {
  handler: startServerAndCreateHandler(server, {
    context: async (args) => createContext(args.req, args.context, args.body),
  }),
});
```

</MultiCodeBlock>

> Your `context` function receives an object with the following properties:

Reviewer: ```suggestion
Your `context` function receives an object with these properties:
```

- `req`: The [HttpRequest](https://learn.microsoft.com/javascript/api/@azure/functions/httprequest) object containing HTTP headers, method, body, query parameters, etc.
- `context`: The [InvocationContext](https://learn.microsoft.com/javascript/api/@azure/functions/invocationcontext) object containing function execution context (trace context, retry context, etc.)
- `body`: The parsed request body

## Deployment

You have several options for deploying your GraphQL API to Azure:

### Using VS Code

1. Install the [Azure Functions extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurefunctions)
2. Right-click your function app in the Azure Functions panel
3. Select "Deploy to Function App"

### Using Azure CLI

> First, build your application:

Reviewer: ```suggestion
1. Build your application:
```
